.github/workflows/main.yml

```diff
@@ -23,6 +23,9 @@ jobs:
           distribution: 'temurin'
           cache: maven
 
+      - name: Install test jar
+        run: mvn install:install-file -pl smart-tests
```

Comment:
I think it should be extracted to the separate shell script, smth like `build-project.sh`, containing `mvn install:install-file -pl smart-tests` and `mvn clean install` with forwarding of all cli arguments to the latter command. Then this script should be used both in GitHub actions workflow and `build-images.sh` script. In this way, we remove the implicit need to run this command before building the project on a new host.